### ISSUE TITLE
problems with packaging an app

### ISSUE THREAD

**Issue by Tealfish** [shoes/shoes4#1182]
when i try to create a package from my app the runs fine in the shoes gui client for mac like so: 
<img width="1122" alt="screen_shot_2015-12-02_at_11_00_10_pm" src="https://cloud.githubusercontent.com/assets/6455171/11554055/70f66c30-9949-11e5-9b49-8616b2efd78b.png">
(btw im on mac osx 10.11) 

it seems to work but when i go to open the app it jumps on my dock then shuts... running the shoes.app>contents>MacOS>JavaAppLauncher in iterm gives me a readout: 
<img width="762" alt="screen_shot_2015-12-02_at_10_57_51_pm" src="https://cloud.githubusercontent.com/assets/6455171/11554085/c7e316ec-9949-11e5-8fe3-258403699fc6.png"> (btw i copied the command and remove the ;exit at the end so i could screen shot) this is my project root
![screen_shot_2015-12-02_at_11_00_44_pm](https://cloud.githubusercontent.com/assets/6455171/11554125/1b150316-994a-11e5-97ea-3676f3ceb314.png)
and my gemfile
<img width="334" alt="screen shot 2015-12-02 at 11 01 50 pm" src="https://cloud.githubusercontent.com/assets/6455171/11554131/254f4044-994a-11e5-8368-66ad77d0eba6.png">
also would this because i commented out path to icon and background in the .yaml file?
thx
Sam

**Comment by PragTob** [comment]
Hi there,

thanks for the report!

Could you please check if [this work-around](https://github.com/shoes/shoes4/issues/1148#issuecomment-161430317) works for you?

Packaging is still a little baby, so sorry for the bumps along the road.

In the future, it'd be cool if you could post the text snippets and not images - easier to copy and past and work with for us :)

Thanks!
Tobi

**Comment by Tealfish** [comment]
yeah np problem for the image text thing and i have yet to try to the work around im at school right now, and thanks for the quick reply.

**Comment by Tealfish** [comment]
running `shoes-swt -p swt:jar ./shoes.rb` gives me 
```
***WARNING: Display must be created on main thread due to Cocoa restrictions.
Ooops, we couldn't start properly. We'll try again.

To avoid this restarting behavior, try one of the following:
  *Use the shoes executable to start your app
  *Add JRUBY_OPTS='-J-XstartOnFirstThread' to your environment before starting.
^C%
```
i used ctrl-c to exit out as it wasent really doing anything the funnything is it made me a .jar file 
'the java jar file "" could not be launched' also it is 3.66GB!?! my script is 60 lines long or so with no images or rich media

**Comment by PragTob** [comment]
Hi there,

that is weird, the J-XstartOnFurstThread should only apply for launching applications, not for packaging them up afaik (double check @jasonrclark )

Also of course it shouldn't be that big. It seems like it is packaging up more. What else is there in the directory where your shoes app is?

From the README:

> The packager will include everything in the directory of your shoes script and below, unless you tell it not to.

So make sure your the directory with shoes.rb doesn't include anything else. You can check what happened if you open the `.jar` file (it is just a ZIP file so your archive manager should be fine).

**Comment by Tealfish** [comment]
the directory is empty gemfile, app.yaml, and shoes.rb opening the app gives me `2015-12-07 5:37:47.972 PM WindowServer[200]: send_datagram_available_ping: pid 216 failed to act on a ping it dequeued before timing out.` in console

**Comment by PragTob** [comment]
yep it is still very strange and I've no idea why/how this happens. Packaging is still unfinished/young and one of the future releases will be dedicated to it, we'll try to sort it out then.

It should not require that switch, however what happens if you run it with the -XStartOnFirstThread flag like the error message suggests?

For every other diagnosis I'd probably need a small test script recreating the problem so I can try it out myself, I only have Linux and Windows though, although @jasonrclark usually covers the Mac :)

**Comment by jasonrclark** [comment]
Been a really long time on this, but the initial reported problem was fixed back in `pre8` with #1345. I'm going to close this out, but @Tealfish if you did have any interest to try again with `pre9` (recently released!) feel free to report back any further issues.
